encoding.c

```diff
@@ -1015,13 +1015,22 @@ rb_enc_get(VALUE obj)
     return rb_enc_from_index(rb_enc_get_index(obj));
 }
 
+const char *
+rb_enc_inspect_name(rb_encoding *enc)
+{
+    if (enc == global_enc_ascii) {
+        return "ASCII-8BIT (BINARY)";
```

Comment:
Why not just `"BINARY"`?
It's an existing alias of it, so it seems fair to use that name in error messages.
```
(irb):8:in `+': incompatible character encodings: UTF-8 and BINARY (Encoding::CompatibilityError)
```
seems much clearer than
```
(irb):8:in `+': incompatible character encodings: UTF-8 and ASCII-8BIT (BINARY) (Encoding::CompatibilityError)
```

Reply:
Because that's what was accepted on the ticket.

Reply:
@nurse Would you be OK with `incompatible character encodings: UTF-8 and BINARY (Encoding::CompatibilityError)`?

For Encoding#inspect, I think `"#<Encoding:ASCII-8BIT (BINARY)>"` as fine, as already the case in the PR.
I'm just concerned the exception message `incompatible character encodings: UTF-8 and ASCII-8BIT (BINARY)` does not really solve the confusion of "incompatible UTF-8 and ASCII ???".